pkg/apis/infinispan/v1/infinispan_types.go

```diff
@@ -43,7 +43,8 @@ type EndpointEncryption struct {
 
 // InfinispanServiceContainerSpec resource requirements specific for service
 type InfinispanServiceContainerSpec struct {
-	Storage string `json:"storage"`
+	Storage          *string `json:"storage,optional,omitempty"`
+	EphemeralStorage *bool   `json:"ephemeralStorage,optional,omitempty"`
```

Comment:
As `Storage` is redundant if `EphemeralStorage` is set and because `Storage` is simply a string, we could allow users to just configure `Spec.Service.Containe.Storage=ephemeral`.

WDYT @rigazilla ?

Reply:
I'm not sure that's good idea. I would like to have on additional Storage container with size, ephemeral and class fields. But this will require additional migration steps on upgrade.

Reply:
+1 to having a struct for configuring storage would be a lot more flexible, but like you say I'm not sure how we can handle this migration.

Reply:
honestly I would go for reusing the `Storage` field. I know the semantic become a little confused this way, but do we have a use case that needs a more structured data model atm?

Reply:
We would like to have an additional structure as next PR related to storage class will add next field.
Also Storage field is a little bit confusing as it's a size but implicitly defined.
We can reuse this field but it will brake convention.

Reply:
Also processing single field update will be nightmare and can't be validated with OpenAPI schema. For example ephemeralStorage must be immutable, but stage size can be extended depends of the provided storage class.

Reply:
ok @dmvolod, it seems reasonable, let's keep the `ephemeralStorage` field.